imageio/plugins/pyav.py

```diff
@@ -785,6 +788,8 @@ def close(self) -> None:
                 self._video_stream.close()
             except ValueError:
                 pass  # stream already closed
+            except AttributeError:
+                pass  # close() was removed in newer versions
```

Comment:
If `CodecContext` no longer has a `.close()` method, how is it disposed of or finalized once we are done writing?

Reply:
That's a good question. It seems the resource would be de-allocated when `__dealloc__` is called:
https://github.com/PyAV-Org/PyAV/blob/e0f0ef415b0ec7dfd48e582cf1735eb3bf04829c/av/codec/context.pyx#L227-L228

More reference: https://patchwork.ffmpeg.org/project/ffmpeg/patch/20240201082938.16687-4-anton@khirnov.net/

Reply:
They say to use `del` https://github.com/PyAV-Org/PyAV/pull/1592

Not sure whats the best way to implement that here to account for all the different ways `_video_stream` can be used by `PyAVPlugin`.

Reply:
I see. Indeed if we want to account for all the different usage, it would require more thoughts. Perhaps we could instead restrict the version of pyav that imageio supports for now to fix the current issue, so that people will have more time to work on a more general solution? If so I could modify the PR to just restrict the version of pyav. Thanks!